nucypher/blockchain/eth/actors.py

```diff
@@ -1004,16 +1011,24 @@ def handle_threshold_signing_request(
             condition_lingo, self.condition_provider_manager, context
         )
 
-        signature_share = self.generate_signature_share(signing_request.data_to_sign)
-        return signature_share
+        message_hash, signature = self.generate_signature_share(
+            signing_request.data_to_sign
+        )
+        response = ThresholdSignatureResponse(
+            message_hash=message_hash,
+            signature=signature,
+        )
+        return response
 
-    def generate_signature_share(self, data: bytes) -> bytes:
+    def generate_signature_share(self, data: bytes) -> Tuple[bytes, bytes]:
         """
         Generate a signature share for the given cohort and data.
         Uses the node's TransactingPower to create an Ethereum-compatible signature for the data.
         """
-        signature = self.transacting_power.sign_message(message=data, standardize=False)
-        return bytes(signature)
+        message_hash, signature = self.transacting_power.sign_message(
+            message=data, standardize=False
+        )
+        return message_hash, bytes(signature)
```

Comment:
Do you need to do the same for `message_hash` as you do for `signature`? i.e.

```suggestion
        return bytes(message_hash), bytes(signature)
```

Reply:
✔️